docs/user_guide.md

```diff
@@ -719,6 +721,70 @@ If you are a project admin, you can disable a Robot Account by clicking "Disable
+
+The action that is taken upon receiving a notification from a Harbor project depends on your continuous integration and development processes. For example, by configuring Harbor to send a `POST` request to a webhook listener at an endpoint of your choice, you can trigger a build and deployment of an application whenever there is a change to an image in the repository.
+
+### Supported Events
+
+You can define one webhook endpoint per project. Webhook notifications provide information about events in JSON format and are delivered by `HTTP` or `HTTPS POST` to an existing webhhook endpoint URL that you provide. The following table describes the events that trigger notifications and the contents of each notification.
+
+|Event|Webhook Event Type|Contents of Notification|
+|---|---|---|
+|Push manifest to registry|`IMAGE PUSH`|Repository namespace name, repository name, tags, manifest digest, image name, image size, push time timestamp, username of user who pushed image|
+|Pull manifest from registry|`IMAGE PULL`|Repository namespace name, repository name, manifest digest, image name, pull time timestamp, username of user who pulled image|
+|Delete manifest from registry|`IMAGE DELETE`|Repository namespace name, repository name, manifest digest, image name, image size, delete time timestamp, username of user who deleted image|
+|Upload Helm chart to registry|`CHART PUSH`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of push, username of user who uploaded chart|
+|Download Helm chart from registry|`CHART PULL`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of push, username of user who pulled chart|
+|Delete Helm chart from registry|`CHART DELETE`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of delete, username of user who deleted chart|
+|Image scan completed|`IMAGE SCAN COMPLETED`|Repository namespace name, repository name, tag scanned, image name, number of critical issues, number of major issues, number of minor issues, last scan status, scan completion time timestamp, vulnerability information (CVE ID, description, link to CVE, criticality, URL for any fix), username of user who performed scan|
+|Image scan failed|`IMAGE SCAN FAILED`|Repository namespace name, repository name, tag scanned, image name, error that occurred, username of user who performed scan|
+
+### Webhook Endpoint Recommendations
+
+The endpoint that receives the webhook should ideally meet the following requirements:
+
+- Have a webhook listener that is capable of interpreting the payload and acting upon the information it contains. For example, running a shell script.
+- Support queues and retry configurations, so that failed webhook notifications are queued and retried until the request succeeds or number of retries exceeds the configured maximum.
+
+It is recommended to create a separate endpoint URL for each project. 

```

Comment:
it is our mode now. but i still think it is a little weird if we add this recommendation here.
i believe there is only one endpoint URL for most users.
do you have any opinion? @reasonerjt @xaleeks

Reply:
Deleted this sentence in https://github.com/goharbor/harbor/pull/8869/commits/e1103b753e11b4d6a794d752620d84c3cccff54f. We can restore it if necessary.